### DISCUSSION TITLE
Profile resolution spec and explicit binding ID

### DISCUSSION THREAD

**Discussion by galtm** [usnistgov/OSCAL#1305] [Q&A]
In the profile resolution spec, why does the requirement on [lines 1207-1209](https://github.com/usnistgov/OSCAL/blob/main/src/specifications/profile-resolution/profile-resolution-specml.xml#L1207) say that `by-id` must not point to the control itself?
```
<req level="must" id="req-modify-alter-add-explicit-id">When an add directive is explicitly bound, the value of the
            <src>by-id</src> child MUST correspond to the value of an
              <src>id</src> on an object inside the control, and not the control itself.</req>
```
The first `<add>` element in one of the examples has `by-id` pointing to the same ID as the parent `<alter>` element. It seems to contradict the spec.
https://github.com/usnistgov/OSCAL/blob/main/src/specifications/profile-resolution/profile-resolution-examples/modify-adds_profile.xml

**Comment by david-waltermire** [comment]
@galtm At face value there seems to be no harm in allowing `by-id` to point to the control id. I am not sure what the trade-offs are with allowing/preventing use of `by-id` in this way. Would choosing one option or another making more or less difficult to implement? Are there other considerations?

**Reply by galtm** [reply]
So far, the main consideration I can think of is that the behaviors of the "before" and "after" position values need to be clear. The spec says that in the implicit binding case, "before" behaves like "starting" and "after" behaves like "ending." If you specify `by-id` then you're using explicit binding. It seems rather subtle for the presence of `by-id` to change how "before" and "after" work.

Is it more intuitive to say that "before" and "after" will insert content outside the target if the target is not a control but inside the target (i.e., like "starting" and "ending") if the target is a control?

**Reply by galtm** [reply]
If #1311 is closed by adopting the July 5 comment, and if conversations about dropping the notion of "before" and "after" behaving like "starting" and "ending" take effect in the spec and code, then I think this discussion will be resolved. The resolution in that case would be that `by-id` is allowed to point to the control id.
